agent/app/run.go

```diff
@@ -43,7 +49,14 @@ func Run(arguments []string) int {
 		// timeout of 30s. This is so that we can catch any http timeout and log the
 		// issue within agent logs.
 		// see https://docs.docker.com/engine/reference/builder/#healthcheck
-		return runHealthcheck("http://localhost:51678/v1/metadata", time.Second*25)
+		// if ECS_AGENT_HEALTHCHECK_HOST env var is set, it will override
+		// the healthcheck's localhost ip address inside the ecs-agent container
+		localhost := "localhost"
+		if localhostOverride := os.Getenv(ECS_AGENT_HEALTHCHECK_HOST_ENV_VAR); localhostOverride != "" {
```

Comment:
perhaps it's more convenient to do  
```golang
if localhostOverride, present := os.LookupEnv(ECS_AGENT_HEALTHCHECK_HOST_ENV_VAR); present { ...
```  
That way we would ignore the case where the env variable is set but accidentally left blank